packages/ai-jsx/src/react/jit-ui/mdx.tsx

```diff
@@ -0,0 +1,115 @@
+import * as AI from '../core.js';
+import { ChatCompletion, SystemMessage } from '../../core/completion.js';
+import React from 'react';
+import { collectComponents } from '../completion.js';
+
+/**
+ * Use GPT-4 with this.
+ */
+export function MdxChatCompletion({ children, usageExamples }: { children: AI.Node; usageExamples: React.ReactNode }) {
+  const components = collectComponents(usageExamples);
+  /* prettier-ignore */
+  return <ChatCompletion>
+    <SystemMessage>
+      You are an assistant who can use React components to work with the user. By default, you use markdown. However, if it's useful, you can also mix in the following React components: {Object.keys(components).join(', ')}.
+      All your responses
+      should be in MDX, which is Markdown For the Component Era. Here are instructions for how to use MDX:
+      === Begin instructions
+      {/* Snipped from https://github.com/mdx-js/mdx/blob/main/docs/docs/what-is-mdx.server.mdx. */}
+      MDX allows you to use JSX in your markdown content.
+      You can import components, such as interactive charts or alerts, and embed them
+      within your content.
+      This makes writing long-form content with components a blast.
+
+      More practically MDX can be explained as a format that combines markdown with
+      JSX and looks as follows:
+
+      === Begin example
```

Comment:
I'm not exactly sure how the string is formatted in JSX, but you might need a few `\n`'s here and there otherwise what the model sees is gonna be very different from what you want.

Reply:
Yep – the model seems to understand pretty well, despite the input being garbled.